# Statistics Fundamentals — Practical Notebook

This notebook follows the concepts introduced in `README.md`. It is intended for practice with small datasets, calculations, sampling, frequencies, and basic statistical notation.

## 1. Working with Observations

A dataset can be represented as a list of observations.

In [1]:
study_hours = [2.5, 4.0, 3.0, 5.0, 1.5, 4.5]

print(study_hours)
print("Number of observations:", len(study_hours))

[2.5, 4.0, 3.0, 5.0, 1.5, 4.5]
Number of observations: 6


## 2. Population and Sample

Create a population and select a simple sample from it.

In [2]:
import random

population = list(range(1, 101))
sample = random.sample(population, 10)

print("Population size:", len(population))
print("Sample:", sample)
print("Sample size:", len(sample))

Population size: 100
Sample: [91, 50, 53, 3, 15, 9, 32, 55, 35, 51]
Sample size: 10


## 3. Mean

Calculate the arithmetic mean using the formula:

$$\bar{x} = \frac{\sum_{i=1}^{n}x_i}{n}$$

In [3]:
values = [10, 20, 30, 40, 50]

total = sum(values)
n = len(values)
mean = total / n

print("Total:", total)
print("n:", n)
print("Mean:", mean)

Total: 150
n: 5
Mean: 30.0


## 4. Weighted Mean

Use the weighted mean formula:

$$\bar{x}_w = \frac{\sum_{i=1}^{n}w_i x_i}{\sum_{i=1}^{n}w_i}$$

In [4]:
scores = [80, 70, 90]
weights = [2, 3, 5]

weighted_sum = sum(score * weight for score, weight in zip(scores, weights))
weight_sum = sum(weights)
weighted_mean = weighted_sum / weight_sum

print("Weighted mean:", weighted_mean)

Weighted mean: 82.0


## 5. Range

In [5]:
values = [12, 15, 20, 27, 30]

data_range = max(values) - min(values)
print("Range:", data_range)

Range: 18


## 6. Frequency Distribution

In [6]:
from collections import Counter

scores = [10, 20, 20, 30, 20, 40, 30, 20]
frequency = Counter(scores)

for score, count in sorted(frequency.items()):
    print(score, ":", count)

10 : 1
20 : 4
30 : 2
40 : 1


## 7. Relative Frequency and Percentage

$$R = \frac{f}{n}$$

$$P = \frac{f}{n}\times100$$

In [7]:
f = 20
n = 100

relative_frequency = f / n
percentage = relative_frequency * 100

print("Relative frequency:", relative_frequency)
print("Percentage:", percentage, "%")

Relative frequency: 0.2
Percentage: 20.0 %


## 8. Stratified Sampling Practice

In [8]:
total_sample = 1000

cse = total_sample * 0.60
ece = total_sample * 0.25
mechanical = total_sample * 0.15

print("CSE:", int(cse))
print("ECE:", int(ece))
print("Mechanical:", int(mechanical))
print("Total:", int(cse + ece + mechanical))

CSE: 600
ECE: 250
Mechanical: 150
Total: 1000


## 9. Statistical Notation Practice

For observations $x_1, x_2, \ldots, x_n$, verify the summation and mean manually using Python.

In [9]:
x = [10, 20, 30]

summation = sum(x)
mean = summation / len(x)

print("Σx:", summation)
print("x̄:", mean)

Σx: 60
x̄: 20.0


## 10. Practice Questions

1. Create a list containing 10 observations and calculate its mean.
2. Calculate the range of the same dataset.
3. Create a frequency table using `collections.Counter`.
4. Calculate relative frequency for each unique value.
5. Explain whether each of these is nominal, ordinal, interval, or ratio data: blood group, satisfaction level, Celsius temperature, weight.
6. Create a population of 1,000 values and select a random sample of 50.
7. Calculate the systematic sampling interval when $N=5000$ and $n=250$.
8. Give one example of sampling bias and explain why increasing the sample size may not remove it.